### ISSUE TITLE
Build failure on Mageia 6 x86_64 with Boost-1.61

### ISSUE THREAD

**Issue by barjac** [freeorion/freeorion#670]
Build log attached.

[freeorion.txt](https://github.com/freeorion/freeorion/files/270373/freeorion.txt)

**Comment by Vezzra** [comment]
Sorry for the late reply, most of our team members are currently swamped it seems 😉 

Looking at the build log it seems that you're trying to install FO using a RPM package provided for your Linux distro. This package hasn't been provided and is not maintained by us, where did you get it?

You'll have to contact the provider/maintainer of this package and report this issue to them. If they can't fix it, then they should contact us so we can try to work things out with them.

Anyway, can anyone of our Linux folks maybe chime in and help out here?

**Comment by geoffthemedio** [comment]
The errors in the log are variations of

```
/home/baz/rpmbuild/BUILD/freeorion-0.4.5-rc3/util/SerializeUniverse.cpp:326:133:   required from here
/home/baz/rpmbuild/BUILD/freeorion-0.4.5-rc3/util/../universe/Ship.h:99:5: error: 'virtual Ship::~Ship()' is protected
     ~Ship() {}
     ^
```

That destructor is indeed protected, but that group of classes is also friends with boost::serialization::access which should avoid such issues, and apparently does on most other systems.

Similar issues have been reported with Boost Serialization in the past, eg. 
https://svn.boost.org/trac/boost/ticket/1708
But this is marked as fixed.

So, I'm inclined to say this is not a problem with FreeOrion code, but with the version of Boost you're building against, and probably not something we should be altering FreeOrion code to fix...

**Comment by barjac** [comment]
OK we will leave this for now as we decided it was too late to move from boost-1.60 to boost-1.61 for Mageia 6 which is now in version freeze.
When the dev branch re-opens after release we will probably move to 1.61 and test the build again - if this is a boost bug then it may have been fixed by then.
For your interest there were only 4 build failures (including FO)  with 1.61 out of 253 packages that use boost.
Barry
(Mageia packaging team)
